src/runtime_src/core/common/api/xrt_module.cpp

```diff
@@ -153,6 +139,20 @@ struct patcher
 
   std::vector<patch_info> m_ctrlcode_patchinfo;
 
+  inline static const std::string&
+  section_name_to_string(buf_type bt)
+  {
+    static const std::array<std::string, static_cast<int>(buf_type::buf_type_count)> Section_Name_Array =
+      { ".ctrltext",
+        ".ctrldata",
+        ".preempt_save",
+        ".preempt_restore",
+        ".pdi",
+        ".ctrlpkt.pm"};
+
+    return Section_Name_Array[static_cast<int>(bt)];
```

Comment:
I think std::string_view is the more modern way of doing this these days.   It's efficient and supported by std::string::find.  Needless to say, need to include `<string_view>`.

```suggestion
    static const std::array<std::string_view, static_cast<int>(buf_type::buf_type_count)> section_name_array =
      { ".ctrltext",
        ".ctrldata",
        ".preempt_save",
        ".preempt_restore",
        ".pdi",
        ".ctrlpkt.pm"};

    return section_name_array[static_cast<int>(bt)];
```